SECURITY.md

```diff
@@ -0,0 +1,21 @@
+# Security Policy
+
+## Supported Versions
+
+Use this section to tell people about which versions of your project are
+currently being supported with security updates.
+
+| Version  | Supported                   |
+| -------- | --------------------------- |
+| 0.24.x   | :white_check_mark:          |
+| 0.23.x   | Depending on usage/severity |
+
+## Reporting a Vulnerability
+
+Please report security bugs [via GitHub](https://github.com/hickory-dns/hickory-dns/security/advisories/new).
+
```

Comment:
It would be good to include a statement here reminding people to not use public issues or PRs to report security issues.

Reply:
Yeah, I agree with this.